pkg/k8s/k8s_service.go

```diff
@@ -10,110 +10,80 @@ import (
 	"k8s.io/apimachinery/pkg/util/intstr"
 )
 
-// GetService retrieves a service.
-func GetService(namespace, name string) (*v1.Service, error) {
-	ctx, cancel := context.WithTimeout(context.Background(), timeout)
-	defer cancel()
-
-	if !IsInitialized() {
-		return nil, ErrKnuuNotInitialized
-	}
-	svc, err := Clientset().CoreV1().Services(namespace).Get(ctx, name, metav1.GetOptions{})
+func (c *Client) GetService(ctx context.Context, name string) (*v1.Service, error) {
+	svc, err := c.clientset.CoreV1().Services(c.namespace).Get(ctx, name, metav1.GetOptions{})
 	if err != nil {
 		return nil, ErrGettingService.WithParams(name).Wrap(err)
 	}
 	return svc, nil
 }
 
-// DeployService deploys a service if it does not exist.
-func DeployService(
-	namespace,
+func (c *Client) DeployService(
```

Comment:
We use `Create`, `Get`, and `Delete` for other resources. Should we use the same function naming for all resources to be more consistent?